### ISSUE TITLE
make qthelp generates non-well-formed .qhp file with unescaped characters

### ISSUE THREAD

**Issue by mcepl** [sphinx-doc/sphinx#6097]
**Describe the bug**
When building QTHelp file for documentation of Python 3.7.2 with Sphinx 1.7.6 I get 
this [Python.qhp](https://github.com/sphinx-doc/sphinx/files/2897075/Python.qhp.broken.txt) file, which breaks 
``qthelpgenerator-qt5``:
```
[  603s] + qhelpgenerator-qt5 build/qthelp/Python.qhp -o build/qthelp/Python.qch
[  603s] Error in line 12722: Expected '#' or '[a-zA-Z]', but got '"'.
```
Only when I apply this diff the QT Help builds:
```diff
--- Python.qhp.broken.txt       2019-02-23 18:34:15.439394876 +0100
+++ Python.qhp.fixed.txt        2019-02-23 13:17:44.509129009 +0100
@@ -12719,8 +12719,8 @@
             <keyword name="!=" ref="reference/expressions.html#index-75"/>
             <keyword name="% (percent)" id="percent.%" ref="library/stdtypes.html#index-13"/
>
             <keyword name="% (percent)" id="percent.%" ref="reference/expressions.html#index
-66"/>
-            <keyword name="&amp; (ampersand)" id="ampersand.&" ref="library/stdtypes.html#in
dex-16"/>
-            <keyword name="&amp; (ampersand)" id="ampersand.&" ref="reference/expressions.ht
ml#index-72"/>
+            <keyword name="&amp; (ampersand)" id="ampersand.&amp;" ref="library/stdtypes.htm
l#index-16"/>
+            <keyword name="&amp; (ampersand)" id="ampersand.&amp;" ref="reference/expression
s.html#index-72"/>
             <keyword name="* (asterisk)" id="asterisk.*" ref="library/stdtypes.html#index-13
"/>
             <keyword name="* (asterisk)" id="asterisk.*" ref="reference/expressions.html#ind
ex-63"/>
             <keyword name="**" ref="library/stdtypes.html#index-13"/>
@@ -12735,8 +12735,8 @@
             <keyword name="/ (slash)" id="slash./" ref="reference/expressions.html#index-65"
/>
             <keyword name="//" ref="library/stdtypes.html#index-13"/>
             <keyword name="//" ref="reference/expressions.html#index-65"/>
-            <keyword name="&lt; (less)" id="less.<" ref="library/stdtypes.html#index-7"/>
-            <keyword name="&lt; (less)" id="less.<" ref="reference/expressions.html#index-75
"/>
+            <keyword name="&lt; (less)" id="less.&lt;" ref="library/stdtypes.html#index-7"/
>
+            <keyword name="&lt; (less)" id="less.&lt;" ref="reference/expressions.html#inde
x-75"/>
             <keyword name="&lt;&lt;" ref="library/stdtypes.html#index-16"/>
             <keyword name="&lt;&lt;" ref="reference/expressions.html#index-69"/>
             <keyword name="&lt;=" ref="library/stdtypes.html#index-7"/>
```

**To Reproduce**
see above

**Your project**
https://build.opensuse.org/package/show/devel:languages:python:Factory/python3
[Full build log](https://github.com/sphinx-doc/sphinx/files/2897085/_log.txt)

**Environment info**
- OS: Linux/openSUSE Tumbleweed
- Python version: [e.g. 3.7.2]
- Sphinx version: [e.g. 1.7.4] (yes, working on upgrade)
- Sphinx extensions:  nothing outside of the main Sphinx package
- Extra tools: QTTools 5.12.0
**Additional context**

**Comment by tk0miya** [comment]
Reproduced with following index:
```
.. index::
   & (ampersand)
   < (less)
   > (greater)
```

And I got following qhp file.
```
        <keywords>
            <keyword name="&amp; (ampersand)" id="ampersand.&" ref="index.html#index-0"/>
            <keyword name="&lt; (less)" id="less.<" ref="index.html#index-0"/>
            <keyword name="&gt; (greater)" id="greater.>" ref="index.html#index-0"/>
        </keywords>
```

**Comment by tk0miya** [comment]
Fixed by https://github.com/sphinx-doc/sphinxcontrib-qthelp/pull/2.
It will be available in Sphinx-2.0.
Thank you for reporting!
